presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -636,6 +639,27 @@ private static boolean shouldUseFileSplitsFromInputFormat(InputFormat<?, ?> inpu
         return splitList;
     }
 
+    @VisibleForTesting
+    static void validateFileBuckets(ListMultimap<Integer, LocatedFileStatus> bucketFiles, int partitionBucketCount, String tableName, String partitionName)
+    {
+        if (bucketFiles.isEmpty()) {
+            return;
+        }
+
+        Integer highestBucketNumber = max(bucketFiles.keySet());
+        // validate the bucket number detected from files, fail the query if the highest bucket number detected from file
+        // exceeds the allowed highest number
+        if (highestBucketNumber >= partitionBucketCount) {
+            throw new PrestoException(HIVE_INVALID_BUCKET_FILES, format(
+                    "Hive table '%s' is corrupt. The highest bucket number in the directory (%s) is greater than the highest bucket number " +
+                            "declared (%s) for partition: %s",
+                    tableName,
+                    highestBucketNumber,
+                    partitionBucketCount - 1,
```

Comment:
Remove `- 1` after changing error message

Reply:
I was thinking the same when I was trying to organize the error message, then I chose to make this trade off since I was not able to figure out a better one - how about `Hive table '%s' is corrupt. The highest bucket number in the directory (%s) exceeds the bucket number range defined by the declared bucket count (%s) for partition: %s`

Reply:
one error message example that may look a bit odd, feel free to suggest a better one 
> Hive table 'tableName' is corrupt. The highest bucket number in the directory (9) exceeds the bucket number range defined by the declared bucket count (9) for partition: partitionName

Reply:
I see what you mean. Maybe we could list the range explicitly?

> Hive table 'tableName' is corrupt. The highest bucket number in the directory (9) exceeds the bucket number range (0..8) defined by the declared bucket count (9) for partition: partitionName

Reply:
I merged the PR since this seems good enough for now. If you want to try adding the range or otherwise improving the error message, feel free to send a follow up PR.

Reply:
Yup, that's my concern. To show the bucket range I'll need to do calculation on the `partitionBucketCount`, maybe extra a method will have better readability for the code, what do you think?
BTW, thanks for your time on reviewing the code.

Reply:
The only calculation is `partitionBucketCount - 1` so I don't think we need a method (the range always starts at zero).

Reply:
Yes, my concerns is whether `subtracting 1` from `partitionBucketCount` is easy for people to understand, otherwise i am good to go with `partitionBucketCount - 1` directly.